frontend/src/components/attendances/AttendanceForm.js

```diff
@@ -0,0 +1,222 @@
+import React from 'react'
+import { withFormik } from 'formik'
+import { connect } from 'react-redux'
+import styled from 'styled-components'
+import * as yup from 'yup'
+import * as actions from '../../store/actions'
+import FormControl from '@material-ui/core/FormControl'
+import TextField from '@material-ui/core/TextField'
+import ThingahaFormModal from '../common/ThingahaFormModal'
+import Checkbox from '@material-ui/core/Checkbox'
+
+import FormControlLabel from '@material-ui/core/FormControlLabel'
+
+const FormContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: space-between;
+  min-width: 24rem;
+
+  & .address {
+    font-weight: bold;
+    font-size: 1rem;
+  }
+
+  & .icon {
+    color: ${(props) => props.theme.palette.common.grey};
+  }
+`
+
+const StyledFormControl = styled(FormControl)`
+  width: 100%;
+  margin-bottom: 1rem;
+`
+
+const AttendanceForm = ({
+  visible,
+  setVisible,
+  editingAttendance,
+  values,
+  setFieldValue,
+  setValues,
+  handleChange,
+  errors,
+  validateForm,
+  submitForm,
+  handleSubmit,
+}) => {
+  return (
+    <ThingahaFormModal
+      title={editingAttendance ? 'Edit Student' : 'Add New Student'}
+      open={visible}
+      onClose={() => setVisible(false)}
+      onCancel={() => setVisible(false)}
+      onSubmit={(e) => {
+        submitForm(e)
+      }}
+    >
+      <form onSubmit={handleSubmit}>
+        <FormContainer>
+          <StyledFormControl>
+            <TextField
```

Comment:
Hi @wy0-0yw , I am sorry my mockups might not be clear but what we want here is a select box of student names. Not a Text Box.

So, you will also need to fetch students using students api and then create the select box with options for that using the list of students you get from the students api.

The select box UI is similar to how we did in Extra Funds page (https://github.com/thingaha/thingaha-web/blob/master/frontend/src/components/extraFunds/ExtraFundForm.js#L65) 

But, of course the way we fetch data is a bit different here because in ExtraFunds the list of transfers is already in the json response but here we need to put an extra function call in `useEffect` to fetch students as well as schools.